dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -578,6 +614,44 @@ def test_categorical_merge(self):
         }
         self.assertCountEqual(report_count, expected_dict)
 
+        # Setting up of profile with stop condition not yet met
+        profile_w_stop_cond_1 = CategoricalColumn("merge_stop_condition_test")
+        profile_w_stop_cond_1.max_sample_size_to_check_stop_condition = 12
```

Comment:
should ensure times are correct, but looks like we don't properly do that in this file rn.

The way we've done previously is mocking timeit as a generator. 
Example:
https://github.com/capitalone/DataProfiler/blob/f206af2f8218713f9372942ba7dbb33b57cf3aba/dataprofiler/tests/profilers/test_profile_builder.py#L106

Reply:
added:
https://github.com/capitalone/DataProfiler/issues/806